lib/elixir/lib/code/normalizer.ex

```diff
@@ -364,9 +364,13 @@ defmodule Code.Normalizer do
 
         last_args =
           case last_arg do
-            {:__block__, _, [[{{:__block__, key_meta, _}, _} | _]] = last_args} ->
-              if key_meta[:format] == :keyword do
-                last_args
+            {:__block__, _meta, [[{{:__block__, key_meta, _}, _} | _] = keyword]} ->
+              if key_meta[:format] == :keyword or block_keyword?(keyword) do
+                [
+                  Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
+                    {{:__block__, [format: :keyword] ++ meta, args}, value}
+                  end)
+                ]
               else
                 [last_arg]
```

Comment:
Isn't the `Enum.map` only needed in the `or` case?
Something like

```suggestion
              cond do
                key_meta[:format] == :keyword ->
                  keyword
                block_keyword?(keyword) ->
                  [
                    Enum.map(keyword, fn {{:__block__, meta, args}, value} ->
                      {{:__block__, [format: :keyword] ++ meta, args}, value}
                    end)
                  ]
              true ->
                [last_arg]
```